test/suites/device-mqtt/config_test.go

```diff
@@ -12,44 +12,113 @@ func TestEnvConfig(t *testing.T) {
 
 	t.Run("change service port", func(t *testing.T) {
 		t.Cleanup(func() {
-			utils.SnapUnset(t, deviceMqttSnap, "env.service.port")
-			utils.SnapUnset(t, deviceMqttSnap, "env")
-			utils.SnapRestart(t, deviceMqttSnap)
+			utils.SnapStop(t, deviceMqttSnap)
 		})
 
 		const newPort = "56789"
 
 		// make sure the port is available before using it
 		utils.RequirePortAvailable(t, newPort)
 
+		// set env. and validate the new port comes online
 		utils.SnapStart(t, deviceMqttSnap)
 		utils.SnapSet(t, deviceMqttSnap, "env.service.port", newPort)
-		utils.SnapRestart(t, deviceMqttSnap)
+		utils.SnapRestart(t, deviceMqttService)
 
 		utils.WaitServiceOnline(t, 60, newPort)
+
+		// unset env. and validate the default port comes online
+		utils.SnapUnset(t, deviceMqttSnap, "env.service.port")
+		utils.SnapUnset(t, deviceMqttSnap, "env")
+		utils.SnapRestart(t, deviceMqttService)
+		utils.WaitServiceOnline(t, 60, defaultServicePort)
 	})
 }
 
 func TestAppConfig(t *testing.T) {
 	// start clean
 	utils.SnapStop(t, deviceMqttSnap)
 
+	t.Run("set and unset apps.", func(t *testing.T) {
+		t.Cleanup(func() {
+			utils.SnapStop(t, deviceMqttSnap)
+		})
+
+		const newPort = "1111"
+
+		// make sure the port is available before using it
+		utils.RequirePortAvailable(t, newPort)
+
+		// set apps. and validate the new port comes online
+		utils.SnapStart(t, deviceMqttSnap)
+		utils.SnapSet(t, deviceMqttSnap, "apps.device-mqtt.config.service.port", newPort)
+		utils.SnapRestart(t, deviceMqttService)
+
+		utils.WaitServiceOnline(t, 60, newPort)
+
+		// unset apps. and validate the default port comes online
+		utils.SnapUnset(t, deviceMqttSnap, "apps.device-mqtt.config.service.port")
+		utils.SnapUnset(t, deviceMqttSnap, "config.service.port")
```

Comment:
`unset apps` didn't work here. Only using `unset apps` and `unset config` together will do the expected unset apps' job.

Reply:
Only using `unset config` didn't solve the issue, and a new issue has been discovered: unset a non-existing option, snap will set the option with an empty value. Here is the detail:
```
$ sudo snap set edgex-device-mqtt apps.device-mqtt.config.service.port=1111
$ sudo snap restart edgex-device-mqtt.device-mqtt

$ sudo snap unset edgex-device-mqtt config.service.port
$ sudo snap restart edgex-device-mqtt.device-mqtt

$ snap get edgex-device-mqtt -d
{
    "apps": {
        "device-mqtt": {
            "config": {
                "service": {
                    "port": 1111
                }
            }
        }
    },
     "config": {
         "service": {}
    }
}

msg="Web server starting (localhost:1111)"
```

Reply:
How about `sudo snap set edgex-device-mqtt !config.service.port`?

Reply:
Tried, same effect.

Reply:
same result here, I have added a comment referring to this issue.